tests/e2e/performance/test_pvc_snapshot_performance.py

```diff
@@ -0,0 +1,235 @@
+            pytest.param(
+                *['1']
+            ),
+            pytest.param(
+                *['10']
+            ),
+            pytest.param(
+                *['100']
+            )
+        ]
+    )
+    @pytest.mark.usefixtures(base_setup.__name__)
+    def test_pvc_snapshot_performance(self, teardown_factory, pvc_size):
+        """
+        1. Run I/O on a pod file.
+        2. Calculate md5sum of the file.
+        3. Take a snapshot of the PVC and measure the time of creation.
+        4. Create a new PVC out of that snapshot.
+        5. Restore From the snapshot and measure the time
+        6. Attach a new pod to it.
+        7. Verify that the file is present on the new pod also.
+        8. Verify that the md5sum of the file on the new pod matches
+           with the md5sum of the file on the original pod.
+
+        This scenario run 3 times and report all results
+        Args:
+            interface(str): The type of the interface
+            (e.g. CephBlockPool, CephFileSystem)
+            pvc_factory: A fixture to create new pvc
+            teardown_factory: A fixture to destroy objects
+        """
+        log.info(f"Running IO on pod {self.pod_obj.name}")
+
+        # Calculating the file size as 25% of the PVC size
+        # in the end the PVC will be 75% full
+        file_size = self.pvc_obj.size * 0.25
+        # Change the file size to MB
+        file_size = f'{int(file_size * 1000)}M'
+        file_name = self.pod_obj.name
+
+        snap_yaml = constants.CSI_RBD_SNAPSHOT_YAML
+        if self.interface == constants.CEPHFILESYSTEM:
+            snap_yaml = constants.CSI_CEPHFS_SNAPSHOT_YAML
+
+        create_times = []
+        restore_times = []
+
+        for test_num in range(3):
+            file_name += f'-{test_num}'
+            log.info('Starting IO on the POD')
+            self.pod_obj.run_io(
+                storage_type='fs', size=file_size, fio_filename=file_name
+            )
+
+            # Wait for fio to finish
+            fio_result = self.pod_obj.get_fio_results()
+            err_count = fio_result.get('jobs')[0].get('error')
+            assert err_count == 0, (
+                f"IO error on pod {self.pod_obj.name}. "
+                f"FIO result: {fio_result}"
+            )
+            log.info('IO on the PVC Finished')
+
+            log.info(f"Verified IO on pod {self.pod_obj.name}.")
+
+            # Verify presence of the file
+            file_path = pod.get_file_path(self.pod_obj, file_name)
+            log.info(f"Actual file path on the pod {file_path}")
+            assert pod.check_file_existence(self.pod_obj, file_path), (
+                f"File {file_name} doesn't exist"
+            )
+            log.info(f"File {file_name} exists in {self.pod_obj.name}")
+
+            # Calculate md5sum
+            orig_md5_sum = pod.cal_md5sum(self.pod_obj, file_name)
+
+            # Take a snapshot
+            log.info('Taking snapshot of the PVC')
```

Comment:
Having the pvc name logged here may be helpful while troubleshooting failures

Reply:
fixed.